src/app/plugins/pubsub.plugin.ts

```diff
@@ -1,23 +1,138 @@
+import Constants from '../../common/constants';
 import { Plugin } from '../../common/plugin';
-import { IContainer, IMessage, ChannelType } from '../../common/types';
+import { ChannelType, IContainer, IHttpResponse, IMessage } from '../../common/types';
+import { MessageEmbed } from 'discord.js';
 
 export class PubSubPlugin extends Plugin {
   public name: string = 'Pub Sub Plugin';
-  public description: string = 'Are publix chicken tender subs on sale?';
-  public usage: string = 'pubsub';
-  public pluginAlias = [];
-  public permission: ChannelType = ChannelType.Bot;
+  public description: string = 'Get prices and steamy pictures of the subs you need in your life.';
+  public usage: string = 'pubSub <subName (optional)> | <"list" | "types" (optional)>';
+  public pluginAlias = ['sub', 'subs', 'sandwich', 'samwich', 'sarnie'];
+  public permission: ChannelType = ChannelType.Public;
+  public pluginChannelName: string = Constants.Channels.Public.Food;
+
+  private _API_URL: string = 'https://api.pubsub-api.dev';
+  private _VALID_KEYS: string[] = ['random', 'list', 'types', ''];
+  private _SUBS: string[] = [];
+  private _EMBED_LIST = new MessageEmbed();
+
+  private _SUB_UPD_THRESH: number = 1000 * 60 * 60 * 24; // in ms, one day.
+  private _LAST_UPD_TIME: number = -this._SUB_UPD_THRESH - 1;
 
   constructor(public container: IContainer) {
     super();
+    this._updateList();
+  }
+
+  private _updateList() {
+    this.container.httpService
+      .get(`${this._API_URL}/allsubs/`)
+      .then((response: IHttpResponse) => {
+        const subs = response.data;
+
+        this._SUBS = subs.map((subData: { name: string }) => subData.name);
+      })
+      .catch((err) => this.container.loggerService.warn(err));
   }
 
   public async execute(message: IMessage, args?: string[]) {
-    const state = this.container.storeService.get('pubSubStore').state;
-    if (state?.onSale) {
-      message.reply('Chicken tender subs are on sale :eyes:');
-    } else {
-      message.reply('Chicken tender subs are **NOT** on sale.');
+    const subIn = this._parseInput(args || []);
+
+    // return the og message
+    if (!subIn) {
+      const state = this.container.storeService.get('pubSubStore').state;
+      if (state?.onSale) {
+        message.reply('Chicken tender subs are on sale :eyes:');
+      } else {
+        message.reply('Chicken tender subs are **NOT** on sale.');
+      }
+      return;
+    }
```

Comment:
This is just the og plugin, I didn't want to do anything with it as it's not mine.  So let me know if you want anything doing with it.

Reply:
just remove the old one